src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/CustomRecordTypeWithUndefinedFields.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Tests;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    // Demonstrate the ability to override the behavior when accessing undefined fields.
+    public class CustomRecordTypeWithUndefinedFields : PowerFxTest
+    {
+        private readonly RecordType _originalRecordType;
+        private readonly RecordType _customRecordType;
+        private readonly TestObj _testObj;
+
+        public CustomRecordTypeWithUndefinedFields()
+        {
+            _originalRecordType = RecordType
+                .Empty()
+                .Add(new NamedFormulaType("prop1", FormulaType.Number))
+                .Add(new NamedFormulaType("prop2", FormulaType.Number))
+                .Add(new NamedFormulaType("prop3", FormulaType.Number));
+
+            _customRecordType = new TreatMissingFieldAsUntypedRecordType(_originalRecordType);
+            _testObj = new TestObj
+            {
+                Properties = new Dictionary<string, object>
+                {
+                    { "prop1",  1 },
+                    { "prop2",  "hello" },
+                    { "prop_not_defined_in_type",  3 },
+                }
+            };
+        }
+
+        [Theory]
+        [InlineData("obj.prop1", 1.0)]
+        [InlineData("IsError(obj.prop2)", true)] // type mismatch should fail at runtime
+        [InlineData("IsBlank(obj.missing)", "#exception")] // type checking fails when referencing undefined type.
+        [InlineData("IsBlank(obj.missing.missing)", "#exception")]
+        [InlineData("IsBlank(obj.prop_not_defined_in_type)", "#exception")]
```

Comment:
>#exception [](http://example.com/codeflow?start=62&length=10)

where is this string coming from?

Reply:
Yes, instead of comparing the "#exception" string, can you compare to ex.Message?

Reply:
Sure, let me update. I was following the existing format from [another scenario test ](https://github.com/microsoft/Power-Fx/blob/a38e155bf82024f4c03989c0782506007c61f4a3/src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioDotNetObjectWrapper.cs#L48)in this folder.

Reply:
Updated. I didn't end up using ex.Message because the actual exception message is outside of scope of this test. Let me know if you'd like another approach.